llvm/lib/Target/NVPTX/NVPTXSubtarget.cpp

```diff
@@ -26,24 +27,38 @@ static cl::opt<bool>
     NoF16Math("nvptx-no-f16-math", cl::Hidden,
               cl::desc("NVPTX Specific: Disable generation of f16 math ops."),
               cl::init(false));
+static cl::opt<unsigned>
+    NextSM("nvptx-next-sm", cl::Hidden,
+           cl::desc("NVPTX Specific: Override SM ID for sm_next."),
+           cl::init(90));
```

Comment:
Do we not have a file somewhere  that lists the last CUDA arch and PTX version? I feel like this will get out of date very fast.

Reply:
Supported values live in [llvm/lib/Target/NVPTX/NVPTX.td, ](https://github.com/llvm/llvm-project/blob/cdc193459d90fad83d2eafaccfe03368a9a8a160/llvm/lib/Target/NVPTX/NVPTX.td#L4)

I think setting it to zero, and erroring out if the exp[licit value is not specified when we're targeting `sm_next` would probably be the right thing to do here. Will fix.

Reply:
Patch now requires the option to be specified if we're overriding the SM.